From e6878148c1742b1f78a659503f5e5067501ebf73 Mon Sep 17 00:00:00 2001
From: Sudhakar Panneerselvam <sudhakar.panneerselvam@oracle.com>
Date: Thu, 22 Jul 2021 18:28:34 +0000
Subject: [PATCH 154/155] Assemble: skip devices that don't match uuid instead
 of aborting the assembly.

This fixes '03r0assem' test as assembly fails when looking for specific
uuid among the device list.

Signed-off-by: Sudhakar Panneerselvam <sudhakar.panneerselvam@oracle.com>
Signed-off-by: Jes Sorensen <jsorensen@fb.com>
---
 Assemble.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/Assemble.c b/Assemble.c
index f954b4db..0df46244 100644
--- a/Assemble.c
+++ b/Assemble.c
@@ -331,6 +331,11 @@ static int select_devices(struct mddev_dev *devlist,
 				/* Ignore unrecognised device if looking for
 				 * specific array */
 				goto loop;
+			if (ident->uuid_set)
+				/* ignore unrecognized device if looking for
+				 * specific uuid
+				 */
+				goto loop;
 
 			pr_err("%s has no superblock - assembly aborted\n",
 			       devname);
-- 
2.25.1

